### ISSUE TITLE
Add Auto Correlation Feature Extractor

### ISSUE THREAD

**Issue by hughrawlinson** [meyda/meyda#12]

**Comment by hughrawlinson** [comment]
I believe @jakubfiala is nearly done on this

**Comment by hughrawlinson** [comment]
Removing v3.0 milestone as it isn't ready.

**Comment by nevosegal** [comment]
@jakubfiala, I had a quick look at the autocorrelation function you started implementing (wow how much had changed since then!), and it feels really complicated for a relatively simple feature. As far as I understand it (correct me if I'm wrong), autocorrelation is basically convolution of a signal with itself. However, this is not efficient and people often implement it as: FFT-> point-wise multiplication of spectrum with its complex conjugate -> IFFT. I'll have a look if I can quickly implement this technique in the following days and let you know how it's going.

**Comment by hughrawlinson** [comment]
@nevosegal `jsfft` does actually provide an (undocumented) inverse fft, and we already have a point-wise multiplication function, so this should be *very* easy as a feature extractor. Do you or @jakubfiala want to get it done? I can pick it up if you like, but I'd like some help with verifying that it's correct, as I'm not very familiar with what this feature is used for, and what its characteristics are.

**Comment by jakubfiala** [comment]
yeah, I think I just followed the technique they used in yaafe. Using the ifft would be immensely more efficient for sure.

**Comment by nevosegal** [comment]
I'll try implementing it and we'll see.

**Comment by nevosegal** [comment]
OK, so I've checked and it is possible to compute it using an FFT, and even simpler using the IFFT of a power spectrum (see [this question on stackoverflow](http://stackoverflow.com/a/3950398/2065922)). The only problem we have is that we only only compute the power spectrum on half of the amplitude spectrum (up to the nyquist), which is problematic for ACF. We should consider passing the full amplitude spectrum to the extractors. We'll probably have to change some for loops but other than that I don't see an issue with doing this.

**Comment by hughrawlinson** [comment]
Ok, anything I can do to help?

**Comment by nevosegal** [comment]
Nope, just writing my thoughts atm.. :D thanks
